web/src/redux/slices/TestDefinition.slice.ts

```diff
@@ -30,11 +30,31 @@ export const assertionResultsToDefinitionList = (assertionResults: TAssertionRes
     isDraft: false,
     isDeleted: false,
     selector,
+    originalSelector: selector,
     assertionList: resultList.flatMap(({assertion}) => [assertion]),
   }));
 };
 
-const testDefinitionSlice = createSlice({
+const testDefinitionSlice = createSlice<
+  ITestDefinitionState,
+  {
+    reset: CaseReducer<ITestDefinitionState>;
+    initDefinitionList: CaseReducer<ITestDefinitionState, PayloadAction<{assertionResults: TAssertionResults}>>;
+    addDefinition: CaseReducer<ITestDefinitionState, PayloadAction<{definition: TTestDefinitionEntry}>>;
+    updateDefinition: CaseReducer<
+      ITestDefinitionState,
+      PayloadAction<{definition: TTestDefinitionEntry; selector: string}>
+    >;
+    removeDefinition: CaseReducer<ITestDefinitionState, PayloadAction<{selector: string}>>;
+    revertDefinition: CaseReducer<ITestDefinitionState, PayloadAction<{originalSelector?: string}>>;
+    resetDefinitionList: CaseReducer<ITestDefinitionState>;
+    setAssertionResults: CaseReducer<ITestDefinitionState, PayloadAction<TAssertionResults>>;
+    clearAffectedSpans: CaseReducer<ITestDefinitionState>;
+    setAffectedSpans: CaseReducer<ITestDefinitionState, PayloadAction<string[]>>;
+    setSelectedAssertion: CaseReducer<ITestDefinitionState, PayloadAction<string>>;
+  },
+  'testDefinition'
+>({
```

Comment:
add better typescript typings

Reply:
I'm always up for this. Should this mean that we can drop the types from each of the reducer functions as well?